zk/stages/stage_sequence_execute_utils.go

```diff
@@ -421,3 +422,45 @@ func checkForBadBatch(
 
 	return false, nil
 }
+
+var (
+	LIMIT_128_KB = uint64(128 * 1024)
+)
+
+type BlockDataChecker struct {
+	limit uint64 // amount of bytes
+	bytes []byte
+}
+
+func NewBlockDataChecker() *BlockDataChecker {
+	return &BlockDataChecker{
+		limit: LIMIT_128_KB,
+		bytes: make([]byte, 0),
+	}
+}
+
+// adds bytes amounting to the block data and checks if the limit is reached
+// if the limit is reached, the data is not added, so this can be reused again for next check
+func (bdc *BlockDataChecker) AddBlockStartData(forkId uint16, deltaTimestamp, l1InfoTreeIndex uint32) bool {
+	newBytes := tx.GenerateStartBlockBatchL2Data(forkId, deltaTimestamp, l1InfoTreeIndex)
+
+	if uint64(len(bdc.bytes)+len(newBytes)) > bdc.limit {
+		return true
+	}
+
+	bdc.bytes = append(bdc.bytes, newBytes...)
+	return false
+}
```

Comment:
I think we can simplify this quite a bit and avoid some of the code calling this function.  We only care about the size of the batchL2Data and we know every changeL2Block transaction is always 9 bytes so we can just keep a track of the number of blocks and append `blockCount * 9` to the transaction bytes, which sadly does need to be fully calculated